compiler/src/dotty/tools/dotc/interactive/Interactive.scala

```diff
@@ -166,31 +166,72 @@ object Interactive {
   private def computeCompletions(pos: SourcePosition, path: List[Tree])(implicit ctx: Context): (Int, List[Symbol]) = {
     val completions = Scopes.newScope.openForMutations
 
-    val (completionPos, prefix, termOnly, typeOnly) = path match {
+    /**
+     * Extract basic info about completion location and the kind of symbols to include.
+     *
+     * @param path     The path to the position where completion happens
+     * @param inImport If set, indicates that this is the completion of an import node. When
+     *                 completing imports, both types and terms are always included.
+     * @return The point where to insert completion, whether terms should be included in results,
+     *         whether types should be included, and whether we're completing an import.
+     */
+    def completionInfo(path: List[Tree], inImport: Boolean): (Int, String, Boolean, Boolean, Boolean) = path match {
```

Comment:
That's a lot of stuff you're returning, this should either be a case class or the code should be refactored to avoid having to return things at this point